Build failure: implicit declaration of `drm_clflush_virt_range` on architecture other than x86

This occurs because the header file `<drm/drm_cache.h>` is currently included conditionally, restricted only to x86 architectures:

https://github.com/DisplayLink/evdi/blob/750044ce73cf2eec9178889c4f851f0ff01368b3/module/evdi_painter.c#L23-L25

However, the function `copy_primary_pixels_on_xe` utilizes `drm_clflush_virt_range` and `drm_memcpy_from_wc` (defined in that header), but the function definition itself is not guarded by the same architecture checks:

https://github.com/DisplayLink/evdi/blob/750044ce73cf2eec9178889c4f851f0ff01368b3/module/evdi_painter.c#L178-L209

This mismatch causes the build to fail when targeting non-x86 architectures (such as AArch64), as the compiler cannot find the definitions for the DRM cache functions.

```
> /build/source/module/evdi_painter.c: In function 'copy_primary_pixels_on_xe':
2026-02-09T14:17:28.3342031Z evdi> /build/source/module/evdi_painter.c:196:17: error: implicit declaration of function 'drm_clflush_virt_range' [-Wimplicit-function-declaration]
2026-02-09T14:17:28.3342707Z evdi>   196 |                 drm_clflush_virt_range(src_mapping.vaddr, byte_span);
2026-02-09T14:17:28.3343032Z evdi>       |                 ^~~~~~~~~~~~~~~~~~~~~~
```

Ref: https://github.com/NixOS/nixpkgs/pull/488694

--- a/module/evdi_painter.c
+++ b/module/evdi_painter.c
@@ -20,9 +20,7 @@
 #include <drm/drmP.h>
 #endif
 #include <drm/drm_edid.h>
-#if defined(CONFIG_X86)
 #include <drm/drm_cache.h>
-#endif
 #include "evdi_drm.h"
 #include "evdi_drm_drv.h"
 #include "evdi_cursor.h"

